From: Danny Kukawka <danny.kukawka@web.de>
Subject: fix detection of TabletPCs

This patch fixes the detection of newer TabletPCs which have
a WACOM UBS-Tablet instead of a serial tablet device. 

Signed-off-by: Danny Kukawka <danny.kukawka@web.de>
---
 10-tabletPCs.fdi |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/fdi/policy/10osvendor/10-tabletPCs.fdi b/fdi/policy/10osvendor/10-tabletPCs.fdi
index f1bddde..2c6d562 100644
--- a/fdi/policy/10osvendor/10-tabletPCs.fdi
+++ b/fdi/policy/10osvendor/10-tabletPCs.fdi
@@ -20,6 +20,22 @@
     <match key="input.device.set" exists="true">
       <append key="info.callouts.add" type="strlist">hal-system-setserial</append>	
     </match>
+
+    <!-- handle USB tablets -->
+    <match key="usb_device.vendor_id" int="0x56a">
+      <!-- WACOM Tablets -->
+      <match key="usb_device.product_id" int_outof="0x90;0x93;0x9a">
+	<!-- TabletPCs with USB Wacom Tablets -->
+          <match key="/org/freedesktop/Hal/devices/computer:system.formfactor" compare_ne="laptop">
+	    <!-- set the correct formfactor if not already laptop -->
+	    <merge key="/org/freedesktop/Hal/devices/computer:system.formfactor" type="string">laptop</merge> 
+	  </match>
+	  <!-- set the correct subtype -->
+	  <merge key="/org/freedesktop/Hal/devices/computer:system.formfactor.subtype" type="string">tabletpc</merge>
+        </match>
+      </match>
+    </match>
+
   </device>
 
 </deviceinfo>
